pdfPreviewInMsg/background.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * This example extension is based on work from:
+ *  - e-gaulue
+ *  - rholeczy
+ */
+
+const PREVIEW_WIDTH = 160;
+const PREVIEW_HEIGHT = PREVIEW_WIDTH * 1.5;
+
+
+/**
+ * Generate a data image URL of a scaled down page of a PDF document.
+ *
+ * @param {PDFDocumentProxy} pdf - the PDF document received from pdf.js
+ * @param {Integer} pageNumber - the page number of the PDF document to be rendered
+ *
+ * returns {String} A data URL of the rendered PDF preview
+ */
+async function generatePdfImageUrl(pdf, pageNumber) {
+  // The PDF will be rendered into a temporary HTML canvas element.
+  let canvas = document.createElement("canvas");
+  canvas.width = PREVIEW_WIDTH;
+  canvas.height = PREVIEW_HEIGHT;
+
+  let page = await pdf.getPage(pageNumber);
+  let viewport = await page.getViewport({ scale: 1.0 });
+  let scale = Math.min(
+    canvas.width / viewport.width,
+    canvas.height / viewport.height
+  );
+  let thumbViewPort = await page.getViewport({ scale });
+  let context = canvas.getContext("2d");
+  await page.render({ canvasContext: context, viewport: thumbViewPort })
+    .promise;
+
+  // Extract the image data URL of the rendered PDF from the canvas.
+  return canvas.toDataURL("image/jpeg");
+}
```

Comment:
I moved the temporary canvas creation into the function itself and let it return the data URL.

Reply:
Ok it's a good idea the const at the top of the file : PREVIEW_WIDTH & PREVIEW_HEIGHT